internal/v4/list.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2014 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package v4 // import "gopkg.in/juju/charmstore.v5-unstable/internal/v4"
+
+import (
+	"net/http"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+
+	"gopkg.in/juju/charmstore.v5-unstable/internal/router"
+)
+
+// GET list[?filter=value…][&include=meta][&sort=field[+dir]]
+// https://github.com/juju/charmstore/blob/v4/docs/API.md#get-list
+func (h *ReqHandler) serveList(_ http.Header, req *http.Request) (interface{}, error) {
+	sp, err := parseSearchParams(req)
```

Comment:
We also need to add username and groups to the search params. Perhaps factor out that stuff from ReqHander.serveSearch.